File: Windows端/wschat/Service/ChatService.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Reflection.Metadata;
using System.Text;
using System.Threading.Tasks;
using wschat.Common;
using wschat.Shared.Contact;
using wschat.Shared.Dtos;
using wschat.Shared.Parameters;

namespace wschat.Service
{
    public class ChatService : BaseService<MsgDto>, IChatService
    {
        private readonly HttpRestClient client;
        private readonly string serviceName = "chat";

        public ChatService(HttpRestClient client) : base(client, "Chat")
        {
            this.client = client;   
        }
        public async Task<ApiResponse<ObservableCollection<MsgDto>>> GetGroupMessage()
        {
            BaseRequest request = new BaseRequest();
            request.Method = RestSharp.Method.GET;
            request.Route = $"{serviceName}/getGroupsMessage";
            var groups = new { userId = AppSession.UserId, groupId = AppSession.GroupId };
            request.Parameter = groups;
            return await client.ExecuteAsync<ObservableCollection<MsgDto>>(request);
        }

        public async Task<ApiResponse<ObservableCollection<MsgDto>>> GetPrivateMessage()
        {
            BaseRequest request = new BaseRequest();
            request.Method = RestSharp.Method.GET;
            request.Route = $"{serviceName}/getPrivateMessageByNum";
            var groups = new { userId1 = AppSession.UserId, userId2 = AppSession.FriendId };
            request.Parameter = groups;
            return await client.ExecuteAsync<ObservableCollection<MsgDto>>(request);
        }

        public async Task<ApiResponse> SendGroupMessage(string content)
        {
            BaseRequest request = new BaseRequest();
            request.Method = RestSharp.Method.GET;
            request.Route = $"{serviceName}/sendGroupsMessage";
            var message = new { userId = AppSession.UserId, toGroupId = AppSession.GroupId, msg=content };
            request.Parameter = message;
            return await client.ExecuteAsync(request);
        }

        public async Task<ApiResponse> SendPrivateMessage(string sendMessages)
        {
            BaseRequest request = new BaseRequest();
            request.Method = RestSharp.Method.GET;
            request.Route = $"{serviceName}/sendPrivateMessage";
            var message = new { userId = AppSession.UserId, toUserId = AppSession.FriendId, msg = sendMessages };
            request.Parameter = message;
            return await client.ExecuteAsync(request);
        }
    }
}

File: 后端/websocketChat/src/main/java/org/websocketchat/websocketchat/service/ChatService.java
package org.websocketchat.websocketchat.service;

import org.websocketchat.websocketchat.pojo.LeaveMessage;
import org.websocketchat.websocketchat.pojo.Result;

/**
 * @author ALL
 * @date 2024/4/23
 * @Description
 */
public interface ChatService {

    Result getAllPrivateMessage(Long userId);

    Result getPrivateMessage(Long userId, Long sendUserId);

    Result getGroupsMessage(Long userId, Long groupId);

    Result sendPrivateMessage(Long userId, Long toUserId, String msg);

    Result sendGroupsMessage(Long userId, Long toGroupId, String msg);

    Result searchPrivateMessage(Long userId, Long toUserId, Long nums);

    Result searchGroupsMessage(Long userId, Long groupId, Long nums);

    Result getPrivateMessageBynum(Long userId, Long sendUserId);

    Result leaveMessage(LeaveMessage leaveMessage);

    Result getLeaveMessage(Long messageId);

    Result getAllLeaveMessage(Long userId);
}
